src/DB/Record.php

```diff
@@ -211,4 +216,44 @@ protected function markAsJustUpdated() {
     protected function initChangedAttributes(AfterSaveEvent $event) {
         $this->setChangedAttributes($event->changedAttributes);
     }
+
+    /**
+     * Creates a query instance for `has-one` or `has-many` relation.
+     *
+     * @param string $class the class name of the related record.
+     * @param array $link the primary-foreign key constraint.
+     * @param bool $multiple whether this query represents a relation to more than one record.
+     *
+     * @return ActiveQueryInterface the relational query object.
+     */
+    protected function createRelationQuery($class, $link, $multiple) {
+        $repository = $this->getRepositoryClassNameForRecord($class);
+        if (null === $repository) {
+            return parent::createRelationQuery($class, $link, $multiple);
+        }
+
+        $query = $repository->find()->getQuery();
+
+        $query->primaryModel = $this;
+        $query->link = $link;
+        $query->multiple = $multiple;
+
+        return $query;
+    }
+
+    protected function getRepositoryClassNameForRecord($recordClass): ?EntitiesRepository {
```

Comment:
It is a mistake. At the beginning, the method had to return the repository class name (in this case method name corresponded to the implementation) but then I decided to return the repository object instead of the class name and I forgot to rename the method.  So it returns only the repository object now. I think will be better call it `getRelatedRepository`.

Reply:
I see now. Signature `getRelatedRepository(string $byRecordClass)` is the right choise indeed .